poo/abstraction/en/Abstraction.md

```diff
@@ -0,0 +1,131 @@
+# Abstraction
+
+The **abstraction** idea, is divide a class in parts, to be able to share (via a parameter passed in method for example) **only one** of this part rather the object “in full”. To understand the utility of this principle and how apply, take for example a class Human, since this is a thing that we know better.
+
+
+```java
+
+class Human {
+
+    int air;
+    int satiety;
+    float energy;
+
+
+    void breathe() {
+        air += 10;
+    }
+
+    void eat() {
+        satiety += 5;
+    }
+
+    void move() {
+        System.out.println("Un humain souhaite se déplacer !");
+    }
+
+    void consumeEnergy(float value) {
+        this.energie -= value;
+    }
+}
+```
+
+Now, imagine that we have a class `Hiking` that allows us to organise a walk. A thing of style:
+
+```java
+
+class Hiking {
+
+    List<Human> participants;
+
+    public Hiking() { this.participants = new ArrayList<>(); }
+
+    void addParticipant(Human participant) { this.participants.add(participant); }
+
+    void moveAll() {
+        for(Human human : participants) {
+            human.move();
+            human.consumeEnergy(3);
+        }
+    }
+}
+```
+
+So far so good, we have that we want. This code works perfectly, except, in reality, it contains two major problems.
+
+1. **We restraint the hiking of the objects of type Human**, whereas anything can **walk** and **exhausted** could take a hiking. It’s not really the kind to forbid to elephants and to tractors to participate :(
+2. We ask a whole human but us all we need is their legs (hum). This class hiking, however currently access to the human respiratory system (via `breathe()` ) and to his stomach (via `eat()` ) ! **So, we give too much power to this class Hiking, since it is able to do things who shouldn’t be accessible to it**. <br>
+
+So, how to get around this two problems? 
+
+The **interfaces** come to our rescue (interfaces are great). Let's make a `Walking` interface, which allows to describe something that moves and gets tired:
+
+```java
+interface Walking {
+    void move();
+    void consumeEnergy(float value);
+}
+```
+
+So, now, our class Human will be able to **implement** this super interface, and in reality we add `implements Walking` and put a little `@Override` on their methods `move` and `consumeEnergy` and the turn is played, since this is a behaviour that we had already implemented beforehand, just without the interface.

```

Comment:
```suggestion
So now, our class `Human` will be able to **implement** this awesome interface, and all that's left to do is to add `implements Walking` and put a little `@Override` on our `move` and `consumeEnergy` methods, and that's it, since this is a behavior that we had already implemented beforehand, only without the interface.

```
I kinda rewrote this section, since I couldn't quite understand some parts that you translated. Also, since it is more common to write in American style, I changed `behaviour` to `behavior`